### PR TITLE
Replace default-detekt-config.yml with generated one

### PR DESCRIPTION
**Pull request by Mauin**
This is a follow up of #590 and contains the same changes atm. Let's merge #590 first. Once that is merged this one should be good to go.

This adds an additional output path to the generator and points it to the `detekt-cli/src/main/resources` directory so that the generated config replaces the currently manually maintained one.

As the generated config is sorted by the order of rules how they are specified in the RuleSetProviders the config will get reformatted in this PR.

Due to #590 the config will automatically get updated every time the `detekt-rules` module gets built.

Link to #496

### CODE REVIEW

**Comment by Mauin** [comment]
Added a TODO to the description. I still need to rebase this once we have merged #590

**Comment by arturbosch** [comment]
#590 is merged!

**Comment by Mauin** [comment]
Rebased and updated the config with the generated one 🎉

File: detekt-generator/src/main/kotlin/io/gitlab/arturbosch/detekt/generator/Args.kt [hunk 1]
```diff
@@ -29,8 +34,11 @@ class Args {
 	val inputPath: Path
 		get() = input ?: throw IllegalStateException("Input path was not initialized by jcommander!")
 
-	val outputPath: Path
-		get() = output ?: throw IllegalStateException("Output path was not initialized by jcommander!")
+	val documentationPath: Path
+		get() = documentation ?: throw IllegalStateException("Output path was not initialized by jcommander!")
```
Comment (arturbosch) [inline detekt-generator/src/main/kotlin/io/gitlab/arturbosch/detekt/generator/Args.kt]: You forgot to update the message of the exception

File: detekt-generator/src/main/kotlin/io/gitlab/arturbosch/detekt/generator/Args.kt [hunk 2]
```diff
@@ -29,8 +34,11 @@ class Args {
...
+
+	val configPath: Path
+		get() = config ?: throw IllegalStateException("Output path was not initialized by jcommander!")
```
Comment (arturbosch) [inline detekt-generator/src/main/kotlin/io/gitlab/arturbosch/detekt/generator/Args.kt]: Here too.

**Comment by arturbosch** [comment]
Thanks for the change! Great work!

**Pull request merged**